tools/munin2snmp/munin2snmp.pl

```diff
@@ -0,0 +1,205 @@
+# Handle request
+sub handle_stats {
+    my ($handler, $registration_info, $request_info, $requests) = @_;
+    update_stats;       # Maybe we should do this in a thread...
+    for (my $request = $requests; $request; $request = $request->next()) {
+        $SNMP::use_numeric = 1;
+        my $oid = $request->getOID();
+        my $noid=SNMP::translateObj($oid);
+        if ($request_info->getMode() == MODE_GET) {
+            # For a GET request, we just check the cache
+            if (exists $cache{$noid}) {
+                $request->setValue(ASN_OCTET_STR, "$cache{$noid}");
+            }
+        } 
+        elsif ($request_info->getMode() == MODE_GETNEXT) {
+            # For a GETNEXT, we need to find a best match. This is the
+            # first match strictly superior to the requested OID.
+            my $bestoid = undef;
+            foreach my $currentoid (@cache_oids) {
+                $currentoid = new NetSNMP::OID($currentoid);
+                next if $currentoid <= $oid;
+                $bestoid = $currentoid;
+                last;
+            }
+            if (defined $bestoid) {
+                $SNMP::use_numeric = 1;
+                my $noid=SNMP::translateObj($bestoid);
+                $request->setOID($bestoid);
+                $request->setValue(ASN_OCTET_STR, "$cache{$noid}");
+            }
+        }
+    }
+}
+
+my $agent = new NetSNMP::agent(
+    'Name' => "munin",
+    'AgentX' => 1);
+
+# Register MIB
+$agent->register("munin-stats", $oidbase,
+    \&handle_stats) or die "registration of handler failed!\n";
+
+# Main loop
+$SIG{'INT'} = \&shutdown;
+$SIG{'QUIT'} = \&shutdown;
+my $running = 1;
+while ($running) {
+    $agent->agent_check_and_process(1);
+}
+$agent->shutdown();
+
+sub shutdown {
+    # Shutdown requested
+    $running = 0;
+}
+
+#muninwalk
+my @collect;
+sub munin_fetch {
+    my $SOCKET = IO::Socket::INET -> new (  PeerAddr => $Munin{HOST},
+        PeerPort => $Munin{PORT},
+        Proto   => 'tcp',
+        Timeout => '10',
+        Type    => SOCK_STREAM,
+    ) or die "Cannot create socket - $@\n";
+    my $tmp = <$SOCKET>;
+    while (my $fetch = shift(@_)) {
+        my $obj = '';
+        if ($fetch =~ /^(\w+)(\.)(\w+)$/ ) {
+            ($fetch,$obj) = ($1,$3);
+        }
+        $SOCKET->print("fetch $fetch\n");
+        select($SOCKET);
+        select(STDOUT);
+        while (<$SOCKET> ) {
+            chomp;
+            $_ =~ s/(.value)//;
+            if ($_ =~ /^(.*)(\s)(.+)$/) {
+                if (($1 eq '# Unknown') or ($1 eq '# Bad')) {
+                    push (@collect,$fetch.$delimiter.'NULL');
+                } else {
+                    if (!$obj or ($1 eq $obj)) {
+                        push (@collect,$fetch.".".$1.$delimiter.$3);
+                    }
+                }
+            } else {
+                last;
+            }
+        }
+    }
+    $SOCKET->print("quit\n");
+    $SOCKET->close();
+    return \@collect if @collect;
+}
+
+=head1 NAME
```

Comment:
Seems that the `README.pod` is already having this, and therefore it is redundant. 

It would be nice to choose either, and auto-generate the other.